src/main/java/com/google/api/codegen/nodejs/NodeJSGapicContext.java

```diff
@@ -157,23 +157,47 @@ public String getNamespace(Interface service) {
   }
 
   /**
-   * The name for the module for this vkit module. This assumes that the service's full name will be
-   * in the format of 'google.some.apiname.version.ServiceName', and extracts the 'apiname' and
-   * 'version' part and combine them to lower-camelcased style (like pubsubV1).
+   * The name for the module for this vkit module. This assumes that the "package_name" in the API
+   * config is in the pattern of "apiname.version", and extracts the 'apiname' and 'version' part
+   * and combine them to lower-camelcased style (like pubsubV1).
    */
-  public String getModuleName(Interface service) {
-    List<String> names = Splitter.on(".").splitToList(service.getFullName());
-    return names.get(names.size() - 3) + lowerUnderscoreToUpperCamel(names.get(names.size() - 2));
+  public String getModuleName() {
+    List<String> names = Splitter.on(".").splitToList(getApiConfig().getPackageName());
+    if (names.size() < 2) {
+      return getApiConfig().getPackageName();
+    }
+    return names.get(0) + lowerUnderscoreToUpperCamel(names.get(names.size() - 1));
```

Comment:
Right, it's going to be `apinameV1beta2`, small `b`. However, I don't think this is problematic, both are equally understandable and acceptable. So I'd like to take simpler implementation.